preview-env/comment/action.yml

```diff
@@ -0,0 +1,46 @@
+---
+name: Manage Deployment Result Summary
+description: |
+  Manages a deployment result summary as comment in the current PR based on deployment artifacts.
+  If deployment artifacts have been uploaded, the action upserts the summary.
+  If there are no deployment artifacts, the summary will get deleted.
+  The deployment artifact name pattern: `deployment-status-<runID>-<runAttempt>-*`
+runs:
+  using: composite
+  steps:
+    - name: Download Deployment Status Artifacts
+      uses: actions/download-artifact@v4
+      with:
+        path: deployment_status
+        pattern: deployment-status-${{ github.run_id }}-${{ github.run_attempt }}-*
+    - name: Check for deployment artifacts
+      id: check_artifacts
+      shell: bash
+      run: |
+        if [ -d deployment_status ] && find deployment_status -type f -name '*.md' | grep -q .; then
+          echo "ARTIFACTS_FOUND=true" >> $GITHUB_OUTPUT
+        else
+          echo "ARTIFACTS_FOUND=false" >> $GITHUB_OUTPUT
+        fi
+    - name: Create deployment status summary
+      if: steps.check_artifacts.outputs.ARTIFACTS_FOUND == 'true'
+      shell: bash
+      run: |
+        shopt -s globstar   # enables the `**` wildcard
+        summary="# :rocket: Deployment Results :rocket:\n"
+        summary+=$(cat deployment_status/**/*.md)
+        echo -e "$summary" | tee summary.md
+    - name: Upsert Deployment Result Summary
+      if: steps.check_artifacts.outputs.ARTIFACTS_FOUND == 'true'
+      uses: mshick/add-pr-comment@v2
+      with:
+        message-path: summary.md
+        refresh-message-position: true
+    - name: Delete Summary
+      # this action will cause a warning:
+      # `Unexpected input(s) 'owner', 'repo', 'issue_number', valid inputs are ['route', 'mediaType']`
+      # This is expected and further info can be found here: https://github.com/octokit/request-action?tab=readme-ov-file#warnings
+      if: steps.check_artifacts.outputs.ARTIFACTS_FOUND == 'true'
+      uses: aki77/delete-pr-comments-action@v2
```

Comment:
As discussed, it's indeed better to use a GHA than custom implementation here 👍

Reply:
as discussed afterwards, went with the shell-script approach